src/handlers/setDependencyVersionHandler.ts

```diff
@@ -169,30 +157,30 @@ function getAllVersionsInTree(pomPath: string, gid: string, aid: string): string
     }
 
     function compare(v1: string, v2: string): number {
-        if (v1 < v2) {
-            return 1;
-        } else if (v1 > v2) {
-            return -1;
-        } else {
-            return 0;
+        const indexCut1: number = v1.indexOf("-"); // filter char
+        const indexCut2: number = v2.indexOf("-");
+        if (indexCut1 !== -1) {
+            v1 = v1.substr(0, indexCut1);
+        }
+        if (indexCut2 !== -1) {
+            v2 = v2.substr(0, indexCut2);
+        }
+        const numbers1: number[] = v1.split(".").map(Number);
```

Comment:
what if for snapshot versions? E.g. compare `0.1.0-SNAPSHOT` , `0.1.1` and `0.1.2-SNAPSHOT`. 

A systematical way is using `SemVer` lib to compare semantic versions if applied. 

Please create a new issue to track this bug if you don't want to fix it in this PR.